### PR TITLE
feat(login-page): add password action

### PR DESCRIPTION
**Pull request by kiaking**
Add password action to the `<SLoginPage>`.

It will also show email/password dialog and handle validations. We can await inside `onSubmit` function to show loading spinner on the form.

```ts
const actions = [
  {
    type: 'password',
    async onSubmit(email, password) {
      // The email/password dialog will show loading spinner
      // while this promise is being awaited
      await login(email, password)
    }
  }
]
```

---
Screenshot 2024-07-18 at 13 20 42

Screenshot 2024-07-18 at 13 20 49

### CODE REVIEW

**Review by brc-dd — DISMISSED** [review]

File: lib/components/SLoginPage.vue [hunk 1]
```diff
@@ -15,37 +19,68 @@ export interface CoverPhotographer {
...
+
+export interface ActionPassword {
+  type: 'password'
+  onSubmit(email: string, password: string): Promise<void>
+}
+
+export interface ActionSocial {
   type: 'google'
-  onClick: () => Promise<void>
+  onClick(): Promise<void>
```
Comment (brc-dd) [inline lib/components/SLoginPage.vue:31]: maybe we should make both onSubmit in v4 👀

Comment (kiaking) [reply lib/components/SLoginPage.vue:31]: Yeah good point 👀

**Comment by kiaking** [comment]
Oh wait, forgot to implement error messaging feature when login failed. Gonna add it. Hmmm... return `false` in `onSubmit` and display error I guess.

**Comment by kiaking** [comment]
@brc-dd 
Added error feature.

```ts
const actions = [
  {
    type: 'password',
    onSubmit(email, password) {
      // Show error if it returns false.
      return login(email, password)
    }
  }
]
```

**Review by brc-dd — APPROVED** [review]

**Pull request merged**